cli/src/cli_util.rs

```diff
@@ -808,17 +808,25 @@ Set which revision the branch points to with `jj branch set {branch_name} -r <RE
         if let Some(revision_str) = revision_str.strip_prefix("all:") {
             self.resolve_revset(revision_str)
         } else {
+            let matches_multiple_revisions = || {
+                self.parse_revset(revision_str)
+                    .is_ok_and(|revset_expression| {
+                        self.evaluate_revset(revset_expression)
+                            .is_ok_and(|revset| revset.iter().take(2).count() > 1)
+                    })
+            };
```

Comment:
Perhaps, better way to fix the problem is to extract the post-processing part of `resolve_single_rev()` to function, and pass `bool` or `Option<String>` to append the hint.

It doesn't make much sense to suggest `all:` if the commmit/branch is diverged/conflict for example.

Reply:
I've extracted `resolve_single_rev()` into a separate function with additional `bool` flag.